pyteal/compiler/optimizer/optimizer.py

```diff
@@ -24,10 +25,25 @@ class OptimizeOptions:
             that have no load dependencies elsewhere.
     """
 
-    def __init__(self, *, scratch_slots: bool = False):
-        self.scratch_slots = scratch_slots
+    def __init__(self, *, scratch_slots: bool = False, frame_pointers: bool = False):
```

Comment:
I think it would be a reasonable property to make using `None` or using `OptimizeOptions()` when creating a `CompileOptions()` behave identically.

Right now they won't because `OptimizeOptions()` _always_ has `frame_pointers=False`.

This is also a problem because existing code is likely to use `OptimizeOptions(scratch_slots=True)`, and it would be undesirable to force such code to have `frame_pointers=False`.

Reply:
Reverted to `None` having the same behavior as `OptimizeOptions()`.  Also, introducing `use_frame_pointer(version)` and `optimize_scratch_slots(version)` to provide transparent behavior as well as be backwards compatible to <= v7 compilationn.